ros_buildfarm/templates/__init__.py

```diff
@@ -96,7 +96,9 @@ def expand_template(template_name, data, options=None):
 
         with open(template_path, 'r') as h:
             content = h.read()
+            interpreter.invoke('beforeFile', name=template_name, file=h)
         interpreter.string(content, template_path, locals=data)
+        interpreter.invoke('afterFile')
```

Comment:
Yes, the base class has a lot of empty methods for various hooks: https://github.com/dirk-thomas/empy/blob/0df7f44312071b976d202a3f8a5fd2ce6aacd174/em.py#L971-L1035